docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user
+
+The basic flow of the structure:
+
+![image info](./UmlDiagrams/CommandParser.png)
+
+**Step 1.** The user inputs command. The command is retrieved by `handleUserInput()` in `CommandHandler`.
+
+**Step 2.** `handleUserInput()` will check command type and create a new `CommandParser` object and call `parse()` in `CommandParser`.
+
+**Step 3.** `CommandParser` will check if the input can be parsed into a valid command. If it fails, it will throw exceptions. After parsing the user’s input, `CommandParser` will create a new `Command` object and return it back to `CommandHandler`.
+
+**Step 4.** `CommandHandler` will call `executeCommand()` in its own class to execute the returned command.
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
+
+**Step1.** After recieving the necessary method call, the `CommandHandler` class will self-call executeCommand with both the list of projects as well as a constructor call to AddProjectCommandParser (and a call to the parse method of AddProjectCommandParser) as its parameters
+
+**Step2.** The AddProjectCommandParser object instance is created
+
+**Step3.** The parse method of AddProjectCommandParser is called with `commands` (the user input) as its parameter.
+
+**Step4.** AddProjectCommandParser will self call its `getProjectTitle` method. This will split the user's input and return the correct name for the project
+
+**Step5.** After returning this projectTitle to itself, `AddProjectCommandParser` will call the constructor for a new `AddProjectCommand` object
+
+**Step6.** This `AddProjectCommand` object will use the newly gathered project name to call the `addProjct` command of the `ProjectList` class.
+
+**Step7.** The status of this command completing is returned to `CommandHandler`, and the output is sent to the user.
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+#### Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
```

Comment:
I think that it would be better for the opt frame to be alt since there is an else statement?